src/core/thread/mle_tlvs.hpp

```diff
@@ -392,16 +361,19 @@ class RouteTlv : public Tlv, public TlvInfo<Tlv::kRoute>
     }
 
     /**
-     * This method sets the Link Quality Out value for a given Router index.
+     * This method sets the Route Data (Link Quality In/Out and Route Cost) for a given Router index.
      *
-     * @param[in]  aRouterIndex  The Router index.
-     * @param[in]  aLinkQuality  The Link Quality Out value for a given Router index.
+     * @param[in]  aRouterIndex    The Router index.
+     * @param[in]  aLinkQualityIn  The Link Quality In value.
+     * @param[in]  aLinkQualityOut The Link Quality Out value.
+     * @param[in]  aRouteCost      The Route Cost value.
      *
      */
-    void SetLinkQualityOut(uint8_t aRouterIndex, LinkQuality aLinkQuality)
+    void SetRouteData(uint8_t aRouterIndex, LinkQuality aLinkQualityIn, LinkQuality aLinkQualityOut, uint8_t aRouteCost)
```

Comment:
Should we move this to cpp file?

Reply:
I tried it at first but it actually slightly increases code size. 
I think when it is inline (in header) since we use it with some constants, compiler can optimize it (calculate he constant during compile time).